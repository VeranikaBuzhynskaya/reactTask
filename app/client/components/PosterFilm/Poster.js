import React from 'react';
import './poster.css';

class Poster extends React.Component{
    render(){
        console.log(this.props);
        const info = this.props.info;
        return (
            <div className="poster">
                <div className="imageBlock"></div>
                <div className="film-info">
                    <div className="film-inform">
                        <span>{info.name}</span>
                        <span>{info.releaseDate}</span>
                    </div>
                    <div>
                        <span>{info.genre}</span>
                    </div>
                </div>
            </div>
        );
    }
}

export default Poster;
