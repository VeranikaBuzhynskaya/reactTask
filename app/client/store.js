import { applyMiddleware, createStore }  from "redux";
import thunk from "redux-thunk";
import {rootReducer} from "./reducers/index";

const middleware = applyMiddleware(thunk);

export default (initialState) => {
    return createStore(rootReducer, initialState, middleware)
};