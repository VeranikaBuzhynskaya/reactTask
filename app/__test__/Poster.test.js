import React from 'react';
import { mount, shallow } from 'enzyme';
import { Link, BrowserRouter} from 'react-router-dom';
import { Provider } from "react-redux";
import Poster from '../client/components/PosterFilm/Poster';
import renderer from 'react-test-renderer';

import data from './mockPosterData';
import store from '../client/store';

describe('Poster', () => {
  let posterFilm;

  beforeAll(() => {
    posterFilm = mount(
        <BrowserRouter>
          <Provider store={store}>
                <Poster info ={data} match={data}/>
          </Provider>
        </BrowserRouter>
    );
    debugger;
    console.log('posterFilm', <Poster info ={data}/> );
  })

  it('has correct title', () => {
    debugger;
    expect(posterFilm.find('.film-inform span').first().text())
        .toEqual('Quentin Tarantino: 20 Years of Filmmaking');
  });

  it('has correct raiting', () => {

    expect(posterFilm.find('.genre').text())
        .toEqual('Raiting: 9');
  });

  it('img has correct src', () => {

    expect(posterFilm.find('img').props().src)
      .toEqual('https://image.tmdb.org/t/p/w500/bzrTHS1JrC8BJszpqKHGX4rLnoV.jpg');
  });

  it('has correct release date', () => {

    expect(posterFilm.find('.film-inform span').last().text())
      .toEqual('2012-11-20');
  });

  it('matches snapshot for components Poster', () =>{
    const tree = renderer.create(
      <BrowserRouter>
      <Provider store={store}>
            <Poster info ={data} match={data}/>
      </Provider>
    </BrowserRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
})