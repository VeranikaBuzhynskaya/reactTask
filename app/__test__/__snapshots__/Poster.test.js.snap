// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster matches snapshot for components Poster 1`] = `
<div
  className="poster"
>
  <a
    href="/film/199951"
    onClick={[Function]}
  >
    <img
      className="imageBlock"
      src="https://image.tmdb.org/t/p/w500/bzrTHS1JrC8BJszpqKHGX4rLnoV.jpg"
    />
  </a>
  <div
    className="film-info"
  >
    <div
      className="film-inform"
    >
      <span>
        Quentin Tarantino: 20 Years of Filmmaking
      </span>
      <span>
        2012-11-20
      </span>
    </div>
    <div>
      <span
        className="genre"
      >
        Raiting: 
        9
      </span>
    </div>
  </div>
</div>
`;
